var/spack/repos/builtin/packages/verrou/package.py

```diff
@@ -0,0 +1,117 @@
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/spack/spack
+# Please also see the NOTICE and LICENSE files for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+import glob
+import os
+import sys
+
+
+class Verrou(AutotoolsPackage):
+    """A floating-point error checker.
+
+    Verrou helps you look for floating-point round-off errors in programs. It
+    implements a stochastic floating-point arithmetic based on random rounding:
+    all floating-point operations are perturbed by randomly switching rounding
+    modes. This can be seen as an asynchronous variant of the CESTAC method, or
+    a subset of Monte Carlo Arithmetic, performing only output randomization
+    through random rounding.
+    """
+
+    homepage = "https://github.com/edf-hpc/verrou"
+    url      = "https://github.com/edf-hpc/verrou.git"
+
+    version('master',
+            git='https://github.com/edf-hpc/verrou.git',
+            branch='master')
+
+    version('2.0.0',
+            git='https://github.com/edf-hpc/verrou.git',
+            commit='a614ad695d060abdfcd3cec11022d6f044a0980d')
+    version('1.1.0',
+            git='https://github.com/edf-hpc/verrou.git',
+            commit='d69787e89507519e14f5ec44150c1882930f692e')
+    version('1.0.0',
+            git='https://github.com/edf-hpc/verrou.git',
+            commit='519dd9a83738f72da8446322e9c403e608940b7a')
+    version('0.9.0',
+            git='https://github.com/edf-hpc/verrou.git',
+            commit='ebabfcf7f4e9a02b130e9819ba6c19e0450eb73d')
+
+    resource(name='valgrind-3.13.0',
+             git='git://sourceware.org/git/valgrind.git',
+             commit='1378ec95f22235e0a8c972cf1cd0abef0b9610d4',
+             when='@1.1.0:,master')
+    resource(name='valgrind-3.12.0',
+             git='git://sourceware.org/git/valgrind.git',
+             commit='36923ab298c8d2791d6c86b6cd5c0667c07449f6',
+             when='@1.0.0:1.0.999')
+    resource(name='valgrind-3.10.1',
+             git='git://sourceware.org/git/valgrind.git',
+             commit='cfc3175a6eb132a5600fcf705f3fa326d71483bf',
+             when='@0.9.0:0.9.999')
+
+    conflicts('valgrind')
```

Comment:
I'm not sure if this works. What exactly are you trying to do here? `conflicts()` just checks to see if the `spec` is valid, it isn't like an LMod conflicts statement.

Reply:
Come to think of it, I am actually not sure if Spack already provides a way to do this. Virtual dependencies seem like the wrong tool here, because they are meant are documentation for dependent packages, whereas here what I am trying to provide is documentation for the user ("please do not load these two packages at the same time").

Any thought?

Reply:
What you're looking for is identical to #4071. @alalazo is currently working on a sprint to improve our Modules interface, so I wouldn't be surprised if this feature is added in the next month or two. For now, I would ignore it.